lib/OnyxUtils.ts

```diff
@@ -626,10 +674,21 @@ function keyChanged<TKey extends OnyxKey>(
         removeLastAccessedKey(key);
     }
 
-    // We are iterating over all subscribers to see if they are interested in the key that has just changed. If the subscriber's  key is a collection key then we will
+    // We get the subscribers interested in the key that has just changed. If the subscriber's  key is a collection key then we will
     // notify them if the key that changed is a collection member. Or if it is a regular key notify them when there is an exact match. Depending on whether the subscriber
     // was connected via withOnyx we will call setState() directly on the withOnyx instance. If it is a regular connection we will pass the data to the provided callback.
-    const stateMappingKeys = Object.keys(callbackToStateMapping);
+    // Given the amount of times this function is called we need to make sure we are not iterating over all subscribers every time. On the other hand, we don't need to
+    // do the same in keysChanged, because we only call that function when a collection key changes, and it doesn't happen that often.
+    let stateMappingKeys = onyxKeyToConnectionIDs.get(key);
+
+    if (!stateMappingKeys) {
+        // Getting the collection key from the specific key because only collection keys were stored in the mapping.
+        stateMappingKeys = onyxKeyToConnectionIDs.get(getPureKey(key));
+        if (!stateMappingKeys) {
+            return;
+        }
+    }
+
```

Comment:
also NAB and just code style discussion / how explicit we want to be.

It might be more explicit:
```js
const connectionKey = isCollectionKey(key) ? getPureKey(key) : key;
const stateMappingKeys = onyxKeyToConnectionIDs.get(connectionKey)
if (!stateMappingKeys) {
  return;
}
```

Here we explicitly check for a collection key, instead of implicitly doing that by having a fallback check, which is more verbose

Reply:
Most of the time the key exists and isn't a collection, also when it enters the main loop 
```
for (let i = 0; i < stateMappingKeys.length; i++) {
```
it also checks for `isCollectionKey`, wanna avoid extra computation

Reply:
okay, makes sense. If its also that way for performance reasons, might be worth to add a comment to avoid future regressions

Reply:
Something that I'm just realize, is when sending a key that is `report_1234` it sends to any component listening to that specific key, but it also sends updates to all components listening to `report_`, I'm looking into a way to update my code + keep the performance